runtime/lua/vim/lsp/inlay_hint.lua

```diff
@@ -4,11 +4,17 @@ local ms = require('vim.lsp.protocol').Methods
 local api = vim.api
 local M = {}
 
+---@class vim.lsp.inlay_hint.config
+---@field format_label? fun(label: string|table): string Formatting function for customizing an inlay hint's label
+
 ---@class lsp.inlay_hint.bufstate
 ---@field version? integer
----@field client_hint? table<integer, table<integer, lsp.InlayHint[]>> client_id -> (lnum -> hints)
+---@field client_hints? table<integer, table<integer, lsp.InlayHint[]>> client_id -> (lnum -> hints)
```

Comment:
This is a good change, maybe do in a separate PR to unblock it.

Reply:
Here it is: https://github.com/neovim/neovim/pull/27250 :)